File: GraphicsTest/GraphicsTest/ScaleLine.m
//
//  ScaleLine.m
//  GraphicsTest
//
//  Created by Vlad on 02/04/2016.
//  Copyright (c) 2016 Vlad. All rights reserved.
//

#import "ScaleLine.h"

@interface ScaleLine ()

@property (assign, nonatomic) double multiplier;
@property (assign, nonatomic) double deflection;

@end

@implementation ScaleLine

-(void)scaleFromVirtualMin:(CGFloat)virtMin andMax:(CGFloat)virtMax toRealMin:(CGFloat)realMin andMax:(CGFloat)realMax
{
    self.multiplier = (realMax - realMin) / (virtMax - virtMin);
    self.deflection = realMin - virtMin * self.multiplier;
}

-(CGFloat)realPositionForVirtualPosition:(CGFloat)position
{
    CGFloat basePosition = position * self.multiplier + self.deflection;
    return basePosition;
}

-(CGFloat)realLengthForVirtualLength:(CGFloat)length
{
    return length * self.multiplier;
}

@end

File: SwiftCode/SwiftCode/ScaleLine.swift
//
//  ScaleLine.swift
//  SwiftCode
//
//  Created by Vlad on 31/07/2016.
//  Copyright © 2016 Vlad. All rights reserved.
//

import UIKit

class ScaleLine: NSObject
{

    var multiplier: CGFloat!
    var deflection: CGFloat!

    func setScale(virtMin: CGFloat, virtMax: CGFloat, realMin: CGFloat, realMax: CGFloat)
    {
        self.multiplier = (realMax - realMin) / (virtMax - virtMin)
        self.deflection = realMin - virtMin * self.multiplier
    }

    func realPositionFromVirtualPosition (position: CGFloat) -> CGFloat
    {
        let basePosition = position * self.multiplier + self.deflection
        return basePosition
    }

    func realLengthForVirtualLength(length:CGFloat) -> CGFloat
    {
        return length * self.multiplier;
    }
}
